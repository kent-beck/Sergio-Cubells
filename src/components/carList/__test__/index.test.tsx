import { render, screen } from "@testing-library/react";
import List from "..";
import { Car } from "../../../types";

describe("Renders a ", () => {
  const carList: Car[] = [
    {
      id: "0",
      title: "Coche",
      euroPrice: "29.870 €",
      image: "someImageUrl",
    },
    {
      id: "1",
      title: "Coche",
      euroPrice: "39.170 €",
      image: "someImageUrl",
    },
  ];

  test("Car list", () => {
    const tree = render(<List carList={carList} />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  test("2 cars", () => {
    render(<List carList={carList} />);

    const expectedCars = 2;

    const result = screen.getAllByText("Coche");

    expect(result.length).toBe(expectedCars);
  });
});
