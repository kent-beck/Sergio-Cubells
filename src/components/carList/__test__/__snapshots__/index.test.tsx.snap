// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a  Car list 1`] = `
<DocumentFragment>
  <div
    class="carListContainer"
  >
    <section
      class="carItemSection"
    >
      <div
        class="carItemContent"
      >
        <img
          alt="Image for Coche"
          class="carItemImage"
          src="someImageUrl"
        />
        <main
          class="carItemTextContainer"
        >
          <div
            class="carItemTitle"
          >
            Coche
          </div>
          <div
            class="carItemFrom"
          >
            desde
          </div>
          <div
            class="carItemPrice"
          >
            29.870 €
          </div>
        </main>
      </div>
    </section>
    <section
      class="carItemSection"
    >
      <div
        class="carItemContent"
      >
        <img
          alt="Image for Coche"
          class="carItemImage"
          src="someImageUrl"
        />
        <main
          class="carItemTextContainer"
        >
          <div
            class="carItemTitle"
          >
            Coche
          </div>
          <div
            class="carItemFrom"
          >
            desde
          </div>
          <div
            class="carItemPrice"
          >
            39.170 €
          </div>
        </main>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
